agent-control/src/opamp/remote_config/validators/signature/public_key.rs

```diff
@@ -0,0 +1,108 @@
+use crate::opamp::remote_config::signature::SigningAlgorithm;
+use crate::opamp::remote_config::validators::signature::public_key_fetcher::{
+    KeyData, PubKeyError,
+};
+use crate::opamp::remote_config::validators::signature::verifier::Verifier;
+use base64::Engine;
+use base64::engine::general_purpose::URL_SAFE_NO_PAD;
+use ring::signature::{ED25519, UnparsedPublicKey};
+
+pub struct PublicKey {
+    pub public_key: UnparsedPublicKey<Vec<u8>>,
+    pub key_id: String,
+}
+
+impl PublicKey {
+    pub fn try_new(data: &KeyData) -> Result<Self, PubKeyError> {
+        if data.use_ != "sig" {
+            return Err(PubKeyError::ParsePubKey("Key use is not 'sig'".to_string()));
+        }
+        if data.kty != "OKP" {
+            return Err(PubKeyError::ParsePubKey(
+                "The only supported algorithm is Okp".to_string(),
+            ));
+        }
+
+        if data.crv != "Ed25519" {
+            return Err(PubKeyError::ParsePubKey(
+                "The only supported crv is Ed25519".to_string(),
+            ));
+        }
+
+        // JWKs make use of the base64url encoding as defined in RFC 4648 [RFC4648]. As allowed by Section 3.2 of the RFC,
+        // this specification mandates that base64url encoding when used with JWKs MUST NOT use padding.
+        let decoded_key = URL_SAFE_NO_PAD
+            .decode(data.x.clone())
+            .map_err(|e| PubKeyError::ParsePubKey(e.to_string()))?;
+
+        Ok(PublicKey {
+            key_id: data.kid.to_string(),
+            public_key: UnparsedPublicKey::new(&ED25519, decoded_key),
+        })
+    }
+}
+
+impl Verifier for PublicKey {
+    type Error = PubKeyError;
+
+    fn verify_signature(
+        &self,
+        signing_algorithm: &SigningAlgorithm,
+        msg: &[u8],
+        signature: &[u8],
+    ) -> Result<(), Self::Error> {
+        if signing_algorithm != &SigningAlgorithm::ED25519 {
+            return Err(PubKeyError::ValidatingSignature(
+                "The only supported algorithm is Ed25519".to_string(),
+            ));
+        }
```

Comment:
Considering that for now we are not considering support for other algorithms, I'd say we can remove the `try_from` implementation from here:

https://github.com/newrelic/newrelic-agent-control/blob/f535e6e951af6038496c1e11555ac2f749d56d41/agent-control/src/opamp/remote_config/signature.rs#L100-L110